container.go

```diff
@@ -0,0 +1,595 @@
+	}
+
+	configuration := string(configurationb)
+
+	if hcsCallbacksSupported {
+		createError = hcsCreateComputeSystem(id, configuration, identity, &handle, &resultp)
+	} else {
+		createError = hcsCreateComputeSystemTP5(id, configuration, &handle, &resultp)
+	}
+	container.handle = handle
+
+	createError = processHcsResult(createError, resultp)
+	if createError != nil {
+		if createError == ErrVmcomputeOperationPending {
+			callbackNumber, err = container.registerCallback(hcsNotificationSystemCreateCompleted)
+			if err != nil {
+				return nil, &ContainerError{Container: container, Operation: operation, Err: err}
+			}
+			defer container.unregisterCallback(callbackNumber)
+
+			if err := container.waitForNotificationTimeout(callbackNumber, defaultTimeout); err != nil {
+				return nil, &ContainerError{Container: container, Operation: operation, Err: err}
+			}
+		} else {
+			return nil, &ContainerError{Container: container, Operation: operation, Err: err}
+		}
+	}
+
+	logrus.Debugf(title+" succeeded id=%s handle=%d", id, container.handle)
+	runtime.SetFinalizer(container, closeContainer)
+	return container, nil
+}
+
+// OpenContainer opens an existing container by ID.
+func OpenContainer(id string) (Container, error) {
+	operation := "OpenContainer"
+	title := "HCSShim::" + operation
+	logrus.Debugf(title+" id=%s", id)
+	var (
+		handle  hcsSystem
+		resultp *uint16
+	)
+
+	container := &container{
+		id: id,
+	}
+
+	err := hcsOpenComputeSystem(id, &handle, &resultp)
+	err = processHcsResult(err, resultp)
+	if err != nil {
+		err = &ContainerError{Container: container, Operation: operation, Err: err}
+		logrus.Error(err)
+		return nil, err
+	}
+
+	container.handle = handle
+
+	logrus.Debugf(title+" succeeded id=%s handle=%d", id, handle)
+	runtime.SetFinalizer(container, closeContainer)
+	return container, nil
+}
+
+// Start synchronously starts the container.
+func (container *container) Start() error {
+	operation := "Start"
+	title := "HCSShim::Container::" + operation
+	logrus.Debugf(title+" id=%s", container.id)
+	var (
+		//options string
+		resultp *uint16
+	)
+
+	err := hcsStartComputeSystemTP5(container.handle, nil, &resultp)
+	err = processHcsResult(err, resultp)
+	if err != nil {
+		err = &ContainerError{Container: container, Operation: operation, Err: err}
+		logrus.Error(err)
+		return err
+	}
+
+	logrus.Debugf(title+" succeeded id=%s", container.id)
+	return nil
+}
+
+// Shutdown requests a container shutdown, but it may not actually be shutdown until Wait() succeeds.
+func (container *container) Shutdown() error {
+	operation := "Shutdown"
+	title := "HCSShim::Container::" + operation
+	logrus.Debugf(title+" id=%s", container.id)
+	var (
+		//options string
+		resultp *uint16
+	)
+
+	err := hcsShutdownComputeSystemTP5(container.handle, nil, &resultp)
+	err = processHcsResult(err, resultp)
+	if err != nil {
+		if err == ErrVmcomputeOperationPending {
```

Comment:
The shutdown may complete synchronously, or async. if it is async. In this case, it is async. I need to update the function description to describe this behaviour.